modules/javafx.graphics/src/main/java/javafx/scene/robot/Robot.java

```diff
@@ -0,0 +1,333 @@
+    /**
+     * Releases the specified {@link KeyCode} key.
+     *
+     * @param keyCode the key to release
+     */
+    public void keyRelease(KeyCode keyCode) {
+        peer.keyRelease(keyCode);
+    }
+
+    /**
+     * Types the specified {@link KeyCode} key.
+     * <p>
+     * This is a convenience method that is equivalent to calling
+     * {@link #keyPress(KeyCode)} followed by {@link #keyRelease(KeyCode)}.
+     *
+     * @param keyCode the key to type
+     */
+    public void keyType(KeyCode keyCode) {
+        keyPress(keyCode);
+        keyRelease(keyCode);
+    }
+
+    /**
+     * Returns the current mouse x-position in screen coordinates.
+     *
+     * @return the current mouse x-position
+     */
+    public double getMouseX() {
+        return peer.getMouseX();
+    }
+
+    /**
+     * Returns the current mouse y-position in screen coordinates.
+     *
+     * @return the current mouse y-position
+     */
+    public double getMouseY() {
+        return peer.getMouseY();
+    }
+
+    /**
+     * Returns the current mouse (x, y) screen coordinates as a {@link Point2D}.
+     *
+     * @return the current mouse (x,y) screen coordinates
+     */
+    public Point2D getMousePosition() {
+        return new Point2D(getMouseX(), getMouseY());
+    }
+
+    /**
+     * Moves the mouse to the specified (x,y) screen coordinates.
+     *
+     * @param x screen coordinate x to move the mouse to
+     * @param y screen coordinate y to move the mouse to
+     */
+    public void mouseMove(double x, double y) {
+        peer.mouseMove(x, y);
+    }
+
+    /**
+     * Moves the mouse to the (x,y) screen coordinates specified by the
+     * given {@code location}.
+     *
+     * @param location the (x,y) coordinates to move the mouse to
+     */
+    public final void mouseMove(Point2D location) {
+        if (location == null) {
+            throw new NullPointerException("location cannot be null");
+        }
+        mouseMove(location.getX(), location.getY());
+    }
+
+    /**
+     * Presses the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to press
+     */
+    public void mousePress(MouseButton... buttons) {
+        peer.mousePress(buttons);
+    }
+
+    /**
+     * Releases the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to release
+     */
+    public void mouseRelease(MouseButton... buttons) {
+        peer.mouseRelease(buttons);
+    }
+
+    /**
+     * Clicks the specified {@code MouseButton}s.
+     * <p>
+     * This is a convenience method that is equivalent to calling
+     * {@link #mousePress(MouseButton...)} followed by {@link #mouseRelease(MouseButton...)}.
+     *
+     * @param buttons the mouse buttons to click
+     */
+    public void mouseClick(MouseButton... buttons) {
+        mousePress(buttons);
+        mouseRelease(buttons);
+    }
+
+    /**
+     * Scrolls the mouse wheel by the specified amount. Positive {@code wheelAmt}s
```

Comment:
Why don't we change the argument value into a method name? 
````java
// rather than
mouseWheel(4) // scrolls up by 4
mouseWheel(-4) // scrolls down by 4

// why not have the method be
mouseWheelUp(4) // scrolls up by 4; negative value throws Exception
mouseWheelDown(4) // scrolls down by 4; negative value throws Exception
````

Reply:
@nlisker I think this is what you're saying. Correct me if I'm wrong:
`ScrollEvent` is for users of JavaFX when they are developing their code. 
````java
// Given a property related to scroll events
Property<Double> prop = new SimpleDoubleProperty(0.0);
// we'd expect ScrollEvent to not have 
//   `getPositiveDeltaX()` and `getNegativeDeltaX()` methods
//   because it's inconvenient:
ScrollEvent se = //
prop.set(se.isDeltaXUp() ? se.getPositiveDeltaX() : se.getNegativeDeltaX())
// thus we just combine the two together into a deltaX value in ScrollEvent
//  and its more convenient
prop.set(se.getDeltaX())
````
If that's how we do it with `ScrollEvent`, then we should do the same for Robot.
````java
// this is more convenient..
double/int amt = // I can't recall which type wheelAmount() takes X'-D
robot.wheelAmount(amt)

// ... than this
double/int amt = //
amt > 0 ? robot.wheelAmountUp(amt) : robot.wheelAmountDown(amt);
````
If I have understood you correctly, I can see why that makes sense. However, I think it would be helpful if there was a default method that would forward to the `wheelAmount` method, so that a developer could insure only scrolling up values are done if that's what he requires:
````java
def test_whether_scrolling_down_works() {
 // given code

 // when
 robot.wheelAmountDown(20);

 // then code
 assert_call()
}
````
However, that is probably its own issue/PR.